### DISCUSSION TITLE
Is there any way we can Explicitly  close or dispose connection to database (Cosmosdb)?

### DISCUSSION THREAD

**Discussion by iambipinpaul** [Gremlinq/ExRam.Gremlinq#451] [Q&A] [answered]
We are using `CosmosDb` , some time we see that , It is not able to connect to the database and return error like.

>Unable to connect to the remote server A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

However connecting to same `CosmsoDb` at same time using SQL API works perfectly fine. 

from here we see that , One instance of azure function can listen upto  256 connection 

https://github.com/projectkudu/kudu/wiki/Azure-Web-App-sandbox#numerical-sandbox-limits

Limit name | Description | Free/Shared/Consumption Limit | Basic+ Limit
-- | -- | -- | --
Threads | Number of threads | 512 | Unlimited (VM limit still applies)
Processes | Number of processes | 32 | Unlimited (VM limit still applies)
Connections | Number of bound sockets outstanding | 600 | Unlimited (VM limit still applies)
Named Pipes | Number of named pipes | 128 | 128
Listen Sockets | Number of listen sockets | 256 | 256

what should be the ideal value for the connection pool configuration 
```json 
{ 

"PoolSize": 250, 
"MaxInProcessPerConnection": 150,
"ReconnectionAttempts": 3, 
"ReconnectionBaseDelayInMilliseconds": 1000 

}
```

how can I explicitly close or dispose the active connection ?

Thank you !

**Comment by danielcweber** [comment, marked as answer]
There's no way that I know of of doing this in Gremlin.Net, so there's unfortunately no way in Gremlinq either. Also, I have no opinion or suggestion about the right pool configuration for your use case. Standard has always worked for me.

**Reply by iambipinpaul** [reply]
Okay , Thank you for the answer.

**Reply by CameronVetter** [reply]
What strategy did you use to solve this problem?  I am running into this exact scenario with gremlinq inside of an azure function.
